[v10] Update contents list dash colour to improve contrast

anandamaryon1 opened the pull request:
## Description
Fixes https://github.com/nhsuk/nhsuk-service-manual-community-backlog/issues/508

Open to counter suggestions on how to make the checklist dashes `grey-2`. 

## Checklist

- [ ] Tested against our [testing policy](https://github.com/nhsuk/nhsuk-frontend/blob/master/docs/contributing/testing.md) (Resolution, Browser & Accessibility)
- [x] Follows our [coding standards and style guide](https://github.com/nhsuk/nhsuk-frontend/blob/master/docs/contributing/coding-standards.md)
- [x] CHANGELOG entry

colinrotherham commented:
@anandamaryon1 I've changed the target branch to `support/10.x`

Think we need a (maybe "Fixes") changelog entry?

anandamaryon1 commented:
> @anandamaryon1 I've changed the target branch to `support/10.x`
> 
> ~Think we need a (maybe "Fixes") changelog entry?~
> 
> I've updated the failing unit tests, plus added a changelog entry and new reference images

Appreciated, thank you

colinrotherham commented on packages/nhsuk-frontend/src/nhsuk/core/settings/_colours-applied.scss:
```diff
@@ -107,6 +107,14 @@ $nhsuk-success-colour: nhsuk-colour("green") !default;
+///
+/// @type Colour
+
+$nhsuk-border-dark-colour: nhsuk-colour("grey-2") !default;
```
No problem, sorted it. Merge if you're happy

Thanks

colinrotherham commented on packages/nhsuk-frontend/src/nhsuk/components/contents-list/_index.scss:
@anandamaryon1 What about this comment?

Maybe we used to use a symbol before `border-top`

anandamaryon1 replied:
Well spotted. Agree, it is referring to an [older implementation](https://github.com/nhsuk/nhsuk-frontend/commit/d23969935eb359166046ed8297ef874c49f23bfa#diff-989970a19fff9b59309a3cf64f5c710e2e1bd5bbcf7f70bfbd0e9655e0001bb0). Let's remove.
